go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -49,22 +44,28 @@ func (agent *ActionAgent) Backup(ctx context.Context, concurrency int, logger lo
 	if !allowMaster && currentTablet.Type == topodatapb.TabletType_MASTER {
 		return fmt.Errorf("type MASTER cannot take backup. if you really need to do this, rerun the backup command with -allow_master")
 	}
-
-	tablet, err := agent.TopoServer.GetTablet(ctx, agent.TabletAlias)
-	if err != nil {
-		return err
-	}
-	if !allowMaster && tablet.Type == topodatapb.TabletType_MASTER {
-		return fmt.Errorf("type MASTER cannot take backup. if you really need to do this, rerun the backup command with -allow_master")
-	}
-	originalType := tablet.Type
-
 	engine, err := mysqlctl.GetBackupEngine()
 	if err != nil {
 		return vterrors.Wrap(err, "failed to find backup engine")
 	}
 	builtin, _ := engine.(*mysqlctl.BuiltinBackupEngine)
+	// get Tablet info from topo so that it is up to date
+	tablet, err := agent.TopoServer.GetTablet(ctx, agent.TabletAlias)
+	if err != nil {
+		return err
+	}
+	originalType := tablet.Type
```

Comment:
Should we update originalType in case it changed after we acquire the action lock? Someone else might be holding it for a long time.

Reply:
good point.